app/main.py

```diff
@@ -1,3 +1,78 @@
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int):
+        self.km = km
+
+    def __str__(self):
+        return f"Distance: {self.km} kilometers"
+
+    def __repr__(self):
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other):
+        if not isinstance(other, Distance):
+            other == int or other == float
+            return Distance(self.km + other)
+        else:
+            return Distance(self.km + other.km)
+
+    def __iadd__(self, other):
+        if not isinstance(other, Distance):
+            other == int or other == float
+            self.km += other
+            return self
+        else:
+            self.km += other.km
+            return self
+
+    def __mul__(self, other):
+        if not isinstance(other, Distance):
+            other == int or other == float
+            return Distance(self.km * other)
+        else:
+            return Distance(self.km * other.km)
+
+    def __truediv__(self, other):
+        if not isinstance(other, Distance):
+            other == int or other == float
+            return Distance(round(self.km / other, 2))
+        else:
+            return Distance(round(self.km / other.km, 2))
+
+    def __lt__(self, other):
+        if not isinstance(other, Distance):
+            other == int or other == float
+            return self.km < other
+        else:
+            return self.km < other.km
+
+    def __gt__(self, other):
+        if not isinstance(other, Distance):
+            other == int or other == float
+            return self.km > other
+        else:
+            return self.km > other.km
+
+    def __eq__(self, other):
+        if not isinstance(other, Distance):
+            other == int or other == float
+            return self.km == other
+        else:
+            return self.km == other.km
```

Comment:
It's not correct, better to write like this:
```
        if isinstance(other, Distance):
            return self.km == other.km
        elif isinstance(other, (int, float)):
            return self.km == other
```

Reply:
Implements this in other methods

Reply:
Got it, thanks!